ntoskrnl/fsrtl/unc.c

```diff
@@ -71,7 +71,7 @@ FsRtlpIsDfsEnabled(VOID)
         return TRUE;
     }
 
-    return ((ULONG)KeyQueryOutput.KeyInfo.Data != 1);
+    return ((ULONG)KeyQueryOutput.KeyInfo.Data[0] != 1);
```

Comment:
Data is `UCHAR Data[1];`, so it would need to be either `return (*(PULONG)KeyQueryOutput.KeyInfo.Data != 1);` or fix the structure definition above to be something like `union { KEY_VALUE_PARTIAL_INFORMATION KeyInfo; struct { ULONG Header[3]; ULONG KeyValue; } } KeyQueryOutput;` and `return (KeyQueryOutput.KeyValue != 0);` or ` struct KEY_VALUE_PARTIAL_INFORMATION KeyInfo; UCHAR Fill[3];}` and `PULONG KeyValue = (PULONG)KeyQueryOutput.KeyInfo.Data; return (*KeyValue != 1);`

Reply:
Aha, I see. There are some nasty little things going on here. Well, okay, will do first option. But on little-endian machine my option will work too (BTW, is PPC port still in game?).

Reply:
Using a portable construct would be nice in any case. Because digging each time in the code to check whether the thing will work when you switch the endianness is a pain.